Duplicati/Library/UsageReporter/ReportSetUploader.cs

```diff
@@ -70,23 +73,20 @@ public static Tuple<Task, IWriteChannel<string>> Run()
                         {
                             if (File.Exists(f))
                             {
-                                var req = (HttpWebRequest)WebRequest.Create(UPLOAD_URL);
-                                req.Method = "POST";
-                                req.ContentType = "application/json; charset=utf-8";
-
                                 int rc;
                                 using (var fs = File.OpenRead(f))
                                 {
                                     if (fs.Length > 0)
                                     {
-                                        req.ContentLength = fs.Length;
-                                        var areq = new Library.Utility.AsyncHttpRequest(req);
-
-                                        using (var rs = areq.GetRequestStream())
-                                            Library.Utility.Utility.CopyStream(fs, rs);
+                                        using var request = new HttpRequestMessage(HttpMethod.Post, UPLOAD_URL);
+                  
+                                        request.Content = new StreamContent(fs);
 
-                                        using (var resp = (HttpWebResponse)areq.GetResponse())
-                                            rc = (int)resp.StatusCode;
+                                        using var timeoutToken = new CancellationTokenSource();
+                                        timeoutToken.CancelAfter(HttpContextSettings.ReadWriteTimeout);

```

Comment:
This should be `HttpContextSettings.OperationTimeout` as that is the overall operation timeout.